src/algorithms/douglasrachford.jl

```diff
@@ -0,0 +1,82 @@
+################################################################################
+# Douglas-Rachford splitting iterable
+#
+# [1] Eckstein, Bertsekas "On the Douglas-Rachford Splitting Method and the
+# Proximal Point Algorithm for Maximal Monotone Operators*",
+# Mathematical Programming, vol. 55, no. 1, pp. 293-318 (1989).
+#
+
+using Base.Iterators
+using ProximalAlgorithms: LBFGS
+using ProximalAlgorithms.IterationTools
+using ProximalOperators: Zero
+using LinearAlgebra
+using Printf
+
+struct DRS_iterable{R <: Real, C <: Union{R, Complex{R}}, T <: AbstractArray{C}}
+    f
+    g
```

Comment:
`f` and `g` shouldn't be typed? of course this can be fixed on a later PR.

Reply:
Of course they should. And I should be less lazy :-p